lib/records/P_RecCore.cc

```diff
@@ -491,6 +493,26 @@ RecSetRecordCounter(const char *name, RecCounter rec_counter, RecSourceT source,
   return RecSetRecord(RECT_NULL, name, RECD_COUNTER, &data, nullptr, source, lock, inc_version);
 }
 
+// check the version of the snap file to remove records.snap or not
+static void
+CheckSnapFileVersion(const char *path)
+{
+  std::ifstream f(path, std::ios::binary);
+  if (f.good()) {
+    // get version and compare & remove
+    char data[5];
+    if (!f.read(data, 5)) {
+      return;
+    }
+    if (data[0] != 'V' || data[1] != PACKAGE_VERSION[0] || data[2] != PACKAGE_VERSION[2] || data[3] != PACKAGE_VERSION[4]) {
```

Comment:
You should check for the null character here too, since you are writing it and reading it.